src/mbgl/gfx/drawable_builder.cpp

```diff
@@ -101,6 +110,9 @@ void DrawableBuilder::setTexture(const std::shared_ptr<gfx::Texture2D>& texture,
 }
 
 void DrawableBuilder::addTriangle(int16_t x0, int16_t y0, int16_t x1, int16_t y1, int16_t x2, int16_t y2) {
+    // check and set the current mode
+    if (!checkAndSetMode(Mode::Primitives)) return;
```

Comment:
Should some logging be added in these cases?

Reply:
I moved the mode checking into `DrawableBuilder::Impl` and logged an error there if the expected `mode` doesn't match the `target`.